etc/checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+		"-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+		"http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+<module name="Checker">
+	<!-- Root Checks -->
+	<module name="RegexpHeader">
+		<property name="headerFile" value="etc/checkstyle/checkstyle-header.txt"/>
+		<property name="fileExtensions" value="java"/>
+	</module>
+	<module name="NewlineAtEndOfFile">
+		<property name="lineSeparator" value="lf"/>
+	</module>
+	<module name="TreeWalker">
+		<property name="tabWidth" value="4"/>
+		<!--<module name="LineLength">-->
+		<!--<property name="max" value="120"/>-->
+		<!--</module>-->
+
+		<!-- Annotations -->
+		<!--<module name="AnnotationUseStyle">-->
+		<!--<property name="elementStyle" value="compact"/>-->
+		<!--</module>-->
+		<module name="MissingOverride"/>
+		<module name="PackageAnnotation"/>
+		<module name="AnnotationLocation">
+			<property name="allowSamelineSingleParameterlessAnnotation"
+					  value="false"/>
+		</module>
+
+		<!-- Block Checks -->
+		<module name="EmptyBlock">
+			<property name="option" value="text"/>
+		</module>
+		<!--
+		<module name="LeftCurly"/>
+		<module name="RightCurly">
+			<property name="option" value="alone"/>
+		</module>
+		<module name="NeedBraces"/>
+		<module name="AvoidNestedBlocks"/>
+		-->
+
+		<!-- tabs instead of spaces -->
+		<!--<module name="RegexpSinglelineJava">-->
+		<!--<property name="format" value="^\t* "/>-->
+		<!--<property name="message" value="Indent must use tab characters"/>-->
+		<!--<property name="ignoreComments" value="true"/>-->
+		<!--</module>-->
+
+		<!-- Class Design -->
+		<!--
+		<module name="FinalClass"/>
+		<module name="InterfaceIsType"/>
+		<module name="MutableException"/>
+		<module name="InnerTypeLast"/>
+		<module name="OneTopLevelClass"/>
+		-->
+
+		<!-- Coding -->
+		<!--
+		<module name="CovariantEquals"/>
+		<module name="EmptyStatement"/>
+		<module name="EqualsHashCode"/>
+		<module name="InnerAssignment"/>
+		<module name="SimplifyBooleanExpression"/>
+		<module name="SimplifyBooleanReturn"/>
+		<module name="StringLiteralEquality"/>
+		<module name="NestedForDepth">
+			<property name="max" value="3"/>
+		</module>
+		<module name="NestedIfDepth">
+			<property name="max" value="3"/>
+		</module>
+		<module name="NestedTryDepth">
+			<property name="max" value="3"/>
+		</module>
+		<module name="MultipleVariableDeclarations"/>
+		<module name="RequireThis">
```

Comment:
if you run a full build like clean install, I know it runs the checkstyle goal.

This file have been copied from what we currently use as default on `spring-cloud-stream` and `acceptance-tests`. Ilaya is probably the best person to discuss all the flags that were decided to enable/disable